connector/connect/common/src/main/protobuf/spark/connect/base.proto

```diff
@@ -852,6 +852,10 @@ message FetchErrorDetailsResponse {
 
     // The query context of a SparkThrowable.
     repeated QueryContext query_contexts = 3;
+
+    // Portable error identifier across SQL engines
+    // If null, error class or SQLSTATE is not set.
+    optional string sql_state = 4;
```

Comment:
Shall we remove the optional modifier? please refer https://github.com/apache/spark/pull/43412

Reply:
I think the problem is the protobuf compatibility (e.g., old version of those messages shoudl work). If we make it required, old messages won't work.

Reply:
@HyukjinKwon Thank you for the explanation.